### ISSUE TITLE
Camera position one frame too late

### ISSUE THREAD

**Issue by einarnordengren** [defold/defold-pbr#5]
This is maybe not very relevant to the defold-pbr itself, but it is to the test project.

When calling PBR.set_camera_world(go.get_world_position("/camera")) in the update function, the PBR extension will get the camera position one frame after each render leading to lighting calculations being a bit off when moving the camera.

I had to do two things to fix this: 1) Call PBR.set_camera_world after the camera gets its new position. 2) Use go.get_position instead of go.get_world_position as go.get_world_position is lagging behind one frame. Posting a message and setting the position using go.get_position in on_message worked for me. Maybe there's a better way.

**Comment by Jhonnyg** [comment]
Hm can you open a pr with your fix perhaps? Not sure we have any other option right now

**Comment by einarnordengren** [comment]
I have opened a PR now. It can be a bit tricky to see the changes, but it helps setting the update frequency to something low and moving back and forth with a high speed wasd-camera.

**Comment by Jhonnyg** [comment]
Thanks @einarnordengren - I wonder if this is perhaps fixable by using the new postupdate function we have in the latest defold version? Otherwise I’ll merge your PR

**Comment by einarnordengren** [comment]
Yes, that would probably be a better solution. I'll update and test it within a few days.

**Comment by einarnordengren** [comment]
I've changed to the built-in late_update function now and it works fine.

Regarding the earlier comment about the differences between using get_position and get_world_position, I don't that is much of a problem. If it is, calling go.update_world_transform() should do the trick.
